video/image_writer.c

```diff
@@ -752,10 +733,24 @@ bool write_image(struct mp_image *image, const struct image_writer_opts *opts,
     if (!dst)
         return false;
 
-    bool success = write(&ctx, dst, filename);
-    if (!success)
+    bool success = false;
+    FILE *fp = fopen(filename, overwrite ? "wb" : "wbx");
+    if (!fp && errno == EEXIST) {
+        mp_err(log, "File '%s' already exists!\n", filename);
+        goto done;
+    }
+    if (!fp) {
+        mp_err(log, "Error opening '%s' for writing!\n", filename);
+        goto done;
+    }
```

Comment:
If we want line savings, we could merge those 
```suggestion
    if (!fp) {
        const char err_str = errno == EEXIST ? "File '%s' already exists!\n" 
                                             : "Error opening '%s' for writing!\n";
        mp_err(log, "Error opening '%s' for writing!\n", filename);
        goto done;
    }
```